### ISSUE TITLE
Can't install some extensions behind a corporate proxy

### ISSUE THREAD

**Issue by HeavyStorm** [Microsoft/vscode#9305]
- VSCode Version: 1.4.0-insider
- OS Version: Microsoft Windows [Version 6.1.7601] (Windows 7)

Steps to Reproduce:

1. Open VS Code.
2. CTRL+P → `ext install npm intellisense` (for instance) → In the side menu, click install.
3. A clock icon appears on the extension icon in the left side of the window. 
4. Nothing else happens. 

I can, however, install some extensions like ESLINT. My guess is that some of those extensions are stored on a different address, and my proxy is blocking access to it. However, I can't determine the URL of the extension package so I'm unable to check if this is the case.

Thanks a lot.

**Comment by joaomoreno** [comment]
Did you configure your proxy? https://code.visualstudio.com/docs/setup/setup-overview#_proxy-server-support

**Comment by HeavyStorm** [comment]
Yes, proxy is set on the settings JSON. That's why I can see the list of available extensions, but I can't download them.

**Comment by joaomoreno** [comment]
Are you in China? Does it reproduce after restarting your machine?

All extensions come from the same place, so it doesn't make sense that some install and others don't.

**Comment by HeavyStorm** [comment]
No, I'm in Brazil. My proxy (websense) may be making something funny. Is there a log I can look into?

(my machine isn't the only one affected, everyone on my team is having the same trouble)

**Comment by joaomoreno** [comment]
Can you show me how you configured the proxy settings?

**Comment by HeavyStorm** [comment]
My personal (*user*) `settings.json` contains the following lines:

```javascript
	//-------- HTTP configuration --------
	// The proxy setting to use. If not set will be taken from the http_proxy and https_proxy environment variables
	"http.proxy": "https://bruleme:Accenture02@webproxy:8080",
	"https.proxy": "http://bruleme:Accenture02@webproxy:8080",
	// Whether the proxy server certificate should be verified against the list of supplied CAs.
	"http.proxyStrictSSL": false,
```

**Comment by HeavyStorm** [comment]
There has to be a reason why eslint installs and other stuff doesn't. If the packages comes from the same url, maybe it's the installation process that's failing. Do I need administrative permission to install extensions? Seems silly since I can just copy them, but might be something along these lines.

**Comment by joaomoreno** [comment]
Can you figure out another extension that works? Maybe a theme...?

**Comment by HeavyStorm** [comment]
Tried some themes. As you can see, spacemacs and sourlick won't install, but Markdown Theme Kit did. 

![image](https://cloud.githubusercontent.com/assets/1890900/16962277/4aec3398-4dc7-11e6-8424-6cb90c4a788d.png)

**Comment by joaomoreno** [comment]
Do you get any information at all when opening the Developer Tools console (<kbd>F1</kbd>, `Developer Tools`)?

**Comment by vborzykin** [comment]
@joaomoreno  I have same problem. And I configured the proxy settings by same pattern. But Developer Tools concol is empty (without errors or warnings).
Also when I open View->Problems (Ctrl+Shift+M) on my settings.json, I get a warning message "String does not match the pattern of "^https?://[^:]+(:\d+)?$|^$""

**Comment by joaomoreno** [comment]
@vborzykin You have another problem. What is the value you've put in `http.proxy`?

**Comment by vborzykin** [comment]
@joaomoreno No, pattern warning is second problem.
First problem is "Can't install extensions behind a corporate proxy" with same diagnostics.

my proxy setup is (from user's  settings.json):
    "http.proxyStrictSSL":false,
    "http.proxy": "http://user:pass@webproxy:3128/",
    "https.proxy": "https://user:pass@webproxy:3128/",

**Comment by joaomoreno** [comment]
@vborzykin Absolutely right about the regexp, pushed a fix.

Do you have the same problem as in you can install ESLint, but not other extensions?

**Comment by vborzykin** [comment]
@joaomoreno Yes, I have

**Comment by joaomoreno** [comment]
But you can install ESLint from the product itself (not command line)?

**Comment by vborzykin** [comment]
No, I can't install ESLint

**Comment by joaomoreno** [comment]
OK, you have another problem, since @HeavyStorm can install **some** extensions. Can you create another issue, showing us what exactly you see when you try to install an extension? Maybe a video recording? Thanks.

**Comment by brianary** [comment]
We also use Websense, but it isn't a proxy, it's a decrypting firewall. It functions as a MitM (manager-in-the-middle), which requires adding the Websense local cert as a trusted authority.

For some really dumb reason, VS and VS Code do not seem to use the Windows list of Trusted Authorities, where the certs were placed by group policy. Right now we can choose just not to use any SSL content (most extensions), or disable all SSL checking (`strict-ssl=false` in `.atomrc`), which would be wildly irresponsible. So that's a huge hassle.

We really need either VS Code to honor imported Trusted Authorities from the Windows cert store, or a way to import a trusted cert into VS Code's special, unique snowflake cert store.

**Comment by HeavyStorm** [comment]
@joaomoreno I found another extension that installs successfully: jscs. 

Tomorrow I'll check developer tools.

**Comment by joaomoreno** [comment]
@HeavyStorm The [latest Insiders build](http://code.visualstudio.com/insiders) has better error messages. Do you mind updating to that one?

**Comment by HeavyStorm** [comment]
@joaomoreno I installed the latest Insiders and now all extensions seems to be installing correctly. Was anything changed in the extensions code?

![image](https://cloud.githubusercontent.com/assets/1890900/17139985/16c559b8-531d-11e6-9457-23641fc4efe5.png)

**Comment by joaomoreno** [comment]
Not really... but maybe the fixes in the error handling case actually fixed something.

Glad to see you're unblocked. I'll close this, reopen pending if you see the issue again.

**Comment by brianary** [comment]
I've opened a new issue for my thing, issue #9790 .

**Comment by pendar747** [comment]
i cant install or view any extensions behind a corporate firewall
